web-admin/src/features/bookmarks/selectors.ts

```diff
@@ -0,0 +1,152 @@
+import {
+  createAdminServiceGetProject,
+  createAdminServiceListBookmarks,
+  type V1Bookmark,
+} from "@rilldata/web-admin/client";
+import { getDashboardStateFromUrl } from "@rilldata/web-common/features/dashboards/proto-state/fromProto";
+import {
+  useMetricsView,
+  useMetricsViewTimeRange,
+} from "@rilldata/web-common/features/dashboards/selectors";
+import { useDashboardStore } from "@rilldata/web-common/features/dashboards/stores/dashboard-stores";
+import { timeControlStateSelector } from "@rilldata/web-common/features/dashboards/time-controls/time-control-store";
+import { ResourceKind } from "@rilldata/web-common/features/entity-management/resource-selectors";
+import { prettyFormatTimeRange } from "@rilldata/web-common/lib/time/ranges";
+import { TimeRangePreset } from "@rilldata/web-common/lib/time/types";
+import {
+  createQueryServiceMetricsViewSchema,
+  type V1MetricsViewSpec,
+  type V1StructType,
+} from "@rilldata/web-common/runtime-client";
+import type { QueryClient } from "@tanstack/query-core";
+import type { CreateQueryResult } from "@tanstack/svelte-query";
+import { derived, type Readable } from "svelte/store";
+
+export function useProjectId(orgName: string, projectName: string) {
+  return createAdminServiceGetProject(
+    orgName,
+    projectName,
+    {},
+    {
+      query: {
+        enabled: !!orgName && !!projectName,
+        select: (resp) => resp.project?.id,
+      },
+    },
+  );
+}
```

Comment:
We'll likely use this selector beyond Bookmarks, so mind moving to `features/projects/selectors.ts`?